client/src/main.tsx

```diff
@@ -25,15 +28,27 @@ const router = createBrowserRouter([
         });
 
         const parsedResponse = userProfileSchema.safeParse(response.json());
-        if (!parsedResponse.success) {
-          console.log("Unable to make implicit login");
-          return redirect("/login");
-        }
-        return <Outlet />;
+        if (!parsedResponse.success) throw new Error("Unable to make implicit login");
+
+        if ("errorMessage" in parsedResponse.data) throw new Error("Error");
+
+        useProfileStore.setState(parsedResponse.data);
+        resolve("Implicit login completed");
       } catch (error) {
-        console.log("Unexpected error");
-        return redirect("/login");
+        reject(error);
       }
+    }, 500);
+  });
+};
+
+const router = createBrowserRouter([
+  {
+    path: "/",
+    loader: async () => {
+      const implicitLoginPromise = implicitLogin();
+      return defer({
+        response: implicitLoginPromise,
+      });
     },
     element: <Loading />,
     errorElement: <NotFound />,
```

Comment:
Could you also make the `/search` and `/profile` routes children of the root route?